php-classes/ActiveRecord.class.php

```diff
@@ -2177,10 +2177,20 @@ protected function _setFieldValue($field, $value)
 
         if ($forceDirty || !array_key_exists($columnName, $this->_record) || $this->_record[$columnName] !== $value) {
             if (array_key_exists($columnName, $this->_record)) {
-                $this->_originalValues[$field] = $originalValue;
+                // only store original value if it hasn't been previously set
+                if (!array_key_exists($field, $this->_originalValues)) {
+                    $this->_originalValues[$field] = $originalValue;
+                } elseif ($this->_originalValues[$field] == $value) { // the value was reverted back to it's original value
+                    unset($this->_originalValues[$field]);
+                }
             }
             $this->_record[$columnName] = $value;
-            $this->_isDirty = true;
+            // set record dirty if there are still dirty fields remaining
```

Comment:
```
if (expression) {
    thing = true;
} else {
    thing = false;
}
```
is an antipattern, it's exactly functionally equivalent to
```
thing = expression;
```
it couldn't produce any different result as the current code if you just move the whole `if (...)` expression to the right side of the equal